### PR TITLE
chore: remove & rename icons

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* re-exports the new icon set from figma
* separates symbols from icons (the former is not supposed to be re-colored)
* gives `icon`'s a better template so we can apply fill & stroke with theme colors directly on them
* sets the default fill of an icon to `currentColor`
* removes the `Icon` component
* adds `AccessibleIcon` utility component
* adds documentation for new things
* fixes storybook TS

### How to test it?

* I dont think there's much value in going through the icons, i'd focus on reviewing the other files that have been changed imo.

### Related issue(s)/PR(s)

* resolves DX-1200

### CODE REVIEW

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -43,3 +43,321 @@ This was just a wrapped component around `Flex`. You should use `Flex` instead.
...
+
+Replaced with `UidField` and exported from `@strapi/icons/symbols`
+
+#### Write
+
+Replaced with `Feather`
+
+## Icon has been removed
+
+The `Icon` component has been removed. It's primary function was to allow you to apply theme colors to icons, this is now doable with the actual icon:
```
Comment (jhoward1994) [inline BREAKING_CHANGES.md:355]: Doesn't this contradict this?

"We encourage users to use the `Icon` component to render icons as it gives access to the `theme` object in styled-components."

File: docs/stories/SimpleMenu.stories.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ type Story = StoryObj<typeof SimpleMenu>;
 
 export const Basic = {
   render: () => (
-    <SimpleMenu label="Actions">
+    <SimpleMenu label="Actionssss">
```
Comment (jhoward1994) [inline docs/stories/SimpleMenu.stories.tsx:16]:
```suggestion
    <SimpleMenu label="Actions">
```

Did you mean this?

File: docs/stories/Accordion.stories.tsx [hunk 3]
```diff
@@ -101,6 +101,8 @@ export const Group = {
   render: () => {
     const [expandedID, setExpandedID] = React.useState<string | null>(null);
 
+    console.log(expandedID);
```
Comment (jhoward1994) [inline docs/stories/Accordion.stories.tsx:104]:
```suggestion
```

Remove this?

File: packages/strapi-design-system/src/SimpleMenu/Menu.tsx [hunk 4]
```diff
@@ -28,14 +28,15 @@ const MenuRoot = DropdownMenu.Root;
...
-  ({ size, endIcon = <CarretDown width="0.6rem" height="0.4rem" aria-hidden />, ...props }, ref) => {
+  ({ size, endIcon = <CaretDown width="0.6rem" height="0.4rem" aria-hidden />, ...props }, ref) => {
     return (
       <DropdownMenu.Trigger asChild>
         <Button
           ref={ref}
           type="button"
           variant="ghost"
           endIcon={endIcon}
+          background="pink"
```
Comment (jhoward1994) [inline packages/strapi-design-system/src/SimpleMenu/Menu.tsx:39]: Sorry if I'm pointing out things that are intended but a background="pink" default seems
strange here?

Comment (joshuaellis) [reply packages/strapi-design-system/src/SimpleMenu/Menu.tsx:39]: No you're totally right, i think i was trying to test something out 😂

File: docs/stories/primitives/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { Title, Meta } from '@storybook/blocks';
+
+<Meta title="Primitives/Overview" parameters={{ viewMode: 'docs' }} />
+
+<Title>Overview</Title>
```
Comment (jhoward1994) [inline docs/stories/primitives/overview.mdx:5]: Do we need these files?

`docs/stories/primitives/overview.mdx`
`docs/stories/foundations/overview.mdx`

We end up with empty overview pages in storybook which looks a bit strange

File: docs/stories/foundations/icons/Icons.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
+import { Typography, lightTheme } from '@strapi/design-system';
```
Comment (jhoward1994) [inline docs/stories/foundations/icons/Icons.stories.tsx:1]: Thank you! these new pages will be so useful

File: docs/stories/primitives/overview.mdx (hunk 5 above)
Comment (joshuaellis) [reply docs/stories/primitives/overview.mdx:5]: I'm gonna keep them for now, the storybook in v2 is very much a WIP, we have tickets to amend this sort of stuff, but it's there to start building the shape we want 💪🏼

File: docs/stories/MainNav.stories.tsx [hunk 7]
```diff
@@ -34,36 +34,34 @@ export const Base = {
...
+              <NavLink href="/content" icon={<Cog />}>
                 Settings
               </NavLink>
             </NavSection>
           </NavSections>
           <NavFooter>
-            <NavUser src="https://avatars.githubusercontent.com/u/3874873?v=4" to="/somewhere-i-belong">
-              John Duff
-            </NavUser>
+            <NavUser src="https://avatars.githubusercontent.com/u/3874873?v=4">John Duff</NavUser>
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply docs/stories/MainNav.stories.tsx:64]: I don't think it does... but we'll be removing this component from the DS, so we'll also get rid of the story, it's probs fine to leave 🤓

File: docs/stories/SimpleMenu.stories.tsx (hunk 2 above)
Comment (joshuaellis) [reply docs/stories/SimpleMenu.stories.tsx:16]: I didnt!

**Comment by madhurisandbhor** [comment]
Screenshot 2024-04-22 at 09 55 57
this looks incorrect with new changes, needs width, height mentioned?

**Comment by joshuaellis** [comment]
> Screenshot 2024-04-22 at 09 55 57 this looks incorrect with new changes, needs width, height mentioned?

let's write a bug in JIRA! :)

**Pull request merged**